main/api/ws.js

```diff
@@ -16,6 +16,37 @@ const logTraffic = process.env.LOG_TRAFFIC
 
 const subs = {}
 
+function updateOrigin (payload, origin) {
+  if (!origin) {
+    log.warn(`Received payload with no origin: ${payload.method}`)
+
+    //log.warn(`Received payload with no origin: ${JSON.stringify(payload)}`)
+    return { ...payload, chainId: payload.chainId || '0x1' }
+  }
+  
+  const originId = uuidv5(origin, uuidv5.DNS)
+  const existingOrigin = store('main.origins', originId)
+
+  if (!existingOrigin && !payload.__extensionConnecting) {
+    // the extension will attempt to send messages (eth_chainId and net_version) in order
+    // to connect. we don't want to store these origins as they'll come from every site
+    // the user visits in their browser
+    store.initOrigin(originId, {
+      name: origin,
+      chain: {
+        id: 1,
+        type: 'ethereum'
+      }
+    })
+  }
+  
+  return {
+    ...payload,
+    chainId: payload.chainId || ('0x' + (existingOrigin && existingOrigin.chain.id) || 1).toString(16),
```

Comment:
Literal with optional chain easier to read, could also extract template var to go further
```
chainId: payload.chainId || `0x${(existingOrigin?.chain.id || 1).toString(16)}`
```